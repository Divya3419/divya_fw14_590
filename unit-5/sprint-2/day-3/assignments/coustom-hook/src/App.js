import logo from './logo.svg';
import './App.css';

import GitHubSearch from './component/GitHubSearch';
import useTimer from './component/useTimer';
import { useEffect, useState } from 'react';
import useThrottle from './component/useThrottle';


function App() {
  
  const ready=useTimer(5000)
  const[count,setCount]=useState(0)

  function debouncer(callback,delay){
    let debounceId
    return function(){
      debounceId && clearTimeout(debounceId)
      debounceId=setTimeout(()=>{
callback()
      },delay)

    }
  }

  const handleScroll=()=>{
    setCount((prev)=>prev+1)
  }
  useThrottle(()=>{
    console.log(count)
  },1000)
  useEffect(()=> {
window.addEventListener("scroll",debouncer(handleScroll,1000))
return()=>{
window.removeEventListener("scroll",debouncer(handleScroll,1000))
}
  },[])
    return (
    <div className="App">
      <h2>Assignment-1</h2>
      <h3>{ready ?"Ready":"Not Ready"}</h3>
      
      <h2>Assignment-2</h2>

         <GitHubSearch/>
         <div style={{position:"sticky",top:"100px"}}>
         <h3 > scroll count :{count}</h3>
         </div>
         

    </div>
  );
}

export default App;
