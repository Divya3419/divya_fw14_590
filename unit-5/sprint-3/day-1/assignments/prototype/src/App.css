.App {
  text-align: center;
  
  box-shadow: rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.23) 0px 3px 6px;
  width:30%;
  margin:auto;
  margin-top:100px;
  height:300px;
  background-color:rgb(32, 75, 116);

}
.App input{
  width:30px;
  height:30px;
  margin:5rem 0.2rem;
  text-align: center;
}
.App h2{
  color:white;
}
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
