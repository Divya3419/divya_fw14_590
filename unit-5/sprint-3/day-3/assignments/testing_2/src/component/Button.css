.btn{
    color:white;
    cursor: pointer;
}

.red{
    background-color:red
}

.green{
    background-color:green;
    color:black;
}

.ghost{
border:none
}


.bordered{
    border:2px solid black
}