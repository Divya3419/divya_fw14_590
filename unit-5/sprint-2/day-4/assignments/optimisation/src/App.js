import logo from './logo.svg';
import './App.css';
import Todo from './component/Todo';
import SearchBar from './component/SearchBar';
import { useEffect, useState } from 'react';
import countries from "./component/utils/country"

function App() {
  const [query, setQuery] = useState("");
  const [loading, setLoading] = useState(false);
  const [sugg, setSugg] = useState([]);

  useEffect(() => {
    if (query === "") {
      setSugg([]);
    } else {
      let newListOfSugg = countries
        .filter((item) =>
          item.country.toLowerCase()
          .indexOf(query) !== -1 ? true : false
        )
        .map((item) => item.country);
      setSugg(newListOfSugg);
    }
    setTimeout(() => setLoading(false), 1000);
  }, [query]);
//console.log(countries)

  return (
    <div className="App">
      <h1>Search Bar</h1>
      {/* <div>Query is: {query}</div> */}
      <SearchBar
        loading={loading}
        setLoading={setLoading}
        sugg={sugg}
        onChange={(val) => setQuery(val)}
      />
    </div>
  );
}

export default App;
