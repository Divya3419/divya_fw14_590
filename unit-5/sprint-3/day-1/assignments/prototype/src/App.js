import logo from './logo.svg';
import './App.css';
import PinInput from './component/PinInput';
import { useState } from 'react';

function App() {
  const [otp,setOtp]=useState("")
  return ( 
    <div className="App">
    <PinInput
     length={5} 
     onChangefirst={(e)=>{setOtp(e)}}/>
     

<h2>OTP value is : {otp}</h2>

    </div>
    
  );
}

export default App;
