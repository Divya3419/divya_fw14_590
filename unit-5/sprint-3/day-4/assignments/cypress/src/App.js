import logo from './logo.svg';
import './App.css';
import { useState } from 'react';

function App() {
  const[count,setCount]=useState(0)
  return (
    <div className="App">
     <h1>count is {count}</h1>
     <button className="inc_button" onClick={()=>setCount(prev =>prev+1)}>inc</button>
     <button className="dec_button" 
     onClick={()=>setCount(prev =>prev-1)}>dec</button>
    </div>
  );
}

export default App;
