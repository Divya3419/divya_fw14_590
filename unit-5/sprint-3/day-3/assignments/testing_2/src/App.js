import logo from './logo.svg';
import './App.css';
import Button from './component/Button';
import { useState } from 'react';
import Counter from './Counter/Counter';


function App() {
  //const[theme,setTheme]=useState("light")

  return (
    <div className="App">
      {/* <div>Click me</div>
      <h1>current theme is {theme}</h1>
      
      <Button onClick={()=>{
        setTheme(theme==="light" ?"dark":"light")
      }}>
        {/* if change name Click me change Theme*/}
        {/* Change Theme */}
      {/* </Button> */} 
<Counter/>
    </div>
  );
}

export default App;
