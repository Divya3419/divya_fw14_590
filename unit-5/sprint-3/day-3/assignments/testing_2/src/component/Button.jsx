import React from 'react'
import "./Button.css"

const Button = ({children,variant,colorSheme,onClick}) => {
  return (
<button data-testid="Button1" className={`btn ${colorSheme || "red"}  
${variant || "ghost"}`}
    onClick={onClick}>
    {children}
    </button>
  )
}

export default Button
