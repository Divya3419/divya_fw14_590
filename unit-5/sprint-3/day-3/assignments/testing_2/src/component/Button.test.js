import {render,screen,fireEvent} from "@testing-library/react"
//import userEvent from "@testing-library/user-event"
import  renderer  from "react-test-renderer"
import App from "../App"
import Button from "./Button"

describe("Button Tests",()=>{
    test("should have button in dom",()=>{
render(<Button>Click me</Button>)//virtually
// screen.debug(); show whole code
 let button=screen.getByText("Click me")
expect(button).toBeInTheDocument()

})

test("should have Id",()=>{
    render(<App></App>)//virtually
    screen.debug(); //show whole code
     let button=screen.getByTestId("Button1")
    expect(button).toBeInTheDocument();
    
    })

// use it or test
it("div text",()=>{
    render(<App></App>)
    let h1=screen.getByText("Click me")
    expect(h1).toBeInTheDocument()
    
        })

    it("should change theme",()=>{
        render(<App></App>)
        let h1=screen.getByText("current theme is light")
        expect(h1).toHaveTextContent("light")

        let themeButton=screen.getByText("Change Theme")
        fireEvent.click(themeButton)
        expect(h1).toHaveTextContent("dark")
            })

            it("should call the function",()=>{
                const mockfn=jest.fn()
                render(<Button onClick={mockfn}>Click Me</Button>)
           const btn=screen.getByTestId("Button1")
        //    if we put two of function it will fail
           //fireEvent.click(btn)
           fireEvent.click(btn)
           expect(mockfn).toBeCalled()
           expect(mockfn).toBeCalledTimes(1);

            }) 

            // for snapshot install renderer
test("should match snapshot",()=>{
     const tree = renderer.create(<Button colorSheme={"green"} variant={"ghost"}>add counter
                </Button>)
                .toJSON()
                expect(tree).toMatchSnapshot()

            })


})