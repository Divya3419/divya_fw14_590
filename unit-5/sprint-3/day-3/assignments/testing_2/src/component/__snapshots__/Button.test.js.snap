// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests should match snapshot 1`] = `
<button
  className="btn green  
ghost"
  data-testid="Button1"
>
  add counter
</button>
`;
